Peer dependency `next` is set to fixed version

### Environment

  ```
System:
  OS: Windows 10 10.0.19043
  CPU: (8) x64 Intel(R) Core(TM) i7-10510U CPU @ 1.80GHz
  Memory: 1.61 GB / 15.75 GB
Binaries:
  Node: 16.17.0 - C:\Program Files\nodejs\node.EXE
  Yarn: 1.22.17 - ~\AppData\Roaming\npm\yarn.CMD
  npm: 8.15.0 - C:\Program Files\nodejs\npm.CMD
Browsers:
  Edge: Spartan (44.19041.1266.0), Chromium (105.0.1343.50)
```

### Reproduction URL

Not needed.

### Describe the issue

Package `next-auth` has peer dependency `"next": "12.2.5"`, i.e. fixed version. Therefore, if you use npm v8 (and probably v7), you get `ERESOLVE unable to resolve dependency tree` if you're not using that exact `next` version.

### How to reproduce

Create a Node.js project depending on e.g. `next@12.3.1` and `next-auth@4.12.0` . Try running `npm install` using npm v8.

### Expected behavior

Peer dependency uses semver **range**, i.e. `"next": "^12.2.5"`, and there are no `ERESOLVE unable to resolve dependency tree` npm errors.

--- a/packages/next-auth/package.json
+++ b/packages/next-auth/package.json
@@ -78,7 +78,7 @@
     "uuid": "^8.3.2"
   },
   "peerDependencies": {
-    "next": "12.2.5",
+    "next": "^12.2.5",
     "nodemailer": "^6.6.5",
     "react": "^17.0.2 || ^18",
     "react-dom": "^17.0.2 || ^18"
@@ -119,7 +119,7 @@
     "jest-environment-jsdom": "^28.1.1",
     "jest-watch-typeahead": "^1.1.0",
     "msw": "^0.42.3",
-    "next": "12.2.5",
+    "next": "^12.2.5",
     "postcss": "^8.4.14",
     "postcss-cli": "^9.1.0",
     "postcss-nested": "^5.0.6",

--- a/packages/next-auth/package.json
+++ b/packages/next-auth/package.json
@@ -119,7 +119,7 @@
     "jest-environment-jsdom": "^28.1.1",
     "jest-watch-typeahead": "^1.1.0",
     "msw": "^0.42.3",
-    "next": "^12.2.5",
+    "next": "12.2.5",
     "postcss": "^8.4.14",
     "postcss-cli": "^9.1.0",
     "postcss-nested": "^5.0.6",

